homeassistant/components/media_player/mpd.py

```diff
@@ -181,13 +183,34 @@ def supported_media_commands(self):
         """Flag of media commands that are supported."""
         return SUPPORT_MPD
 
+    @property
+    def source(self):
+        """Name of the current input source."""
+        return self.currentplaylist
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self.playlists
+
+    def select_source(self, source):
+        """Choose a different available playlist and play it."""
+        self.play_media(MEDIA_TYPE_PLAYLIST, source)
+
     def turn_off(self):
         """Service to send the MPD the command to stop playing."""
         self.client.stop()
 
     def turn_on(self):
         """Service to send the MPD the command to start playing."""
         self.client.play()
+        self._update_playlists()
```

Comment:
Since the intention here is a force-update you could add `no_throttle=True` to the call.

https://github.com/home-assistant/home-assistant/blob/b0e3d5a576f5e00b5a31d3c2a9770a3be383ebaf/homeassistant/util/__init__.py#L235